src/main/java/org/broadinstitute/hellbender/tools/gvs/common/WeightedSplitIntervals.java

```diff
@@ -0,0 +1,224 @@
+    public long defaultWeightPerBase = 0L;
+
+    @Override
+    public void onTraversalStart() {
+        ParamUtils.isPositive(scatterCount, "scatter-count must be > 0.");
+
+        if (!outputDir.exists() && !outputDir.mkdir()) {
+            throw new RuntimeIOException("Unable to create directory: " + outputDir.getAbsolutePath());
+        }
+
+        // in general dictionary will be from the reference, but using -I reads.bam or -F variants.vcf
+        // to use the sequence dict from a bam or vcf is also supported
+        final SAMSequenceDictionary sequenceDictionary = getBestAvailableSequenceDictionary();
+
+        final List<SimpleInterval> inputIntervals = intervalArgumentCollection.getIntervals(sequenceDictionary);
+        final IntervalList intervalsWithWeights = preprocessIntervalsWithWeights(sequenceDictionary, inputIntervals);
+
+        // calculate total weights and target weight per shard
+        float totalWeight = 0;
+        for (Interval i : intervalsWithWeights) {
+            totalWeight += ((WeightedInterval) i).getWeight();
+        }
+        float targetWeightPerScatter = totalWeight / (float) scatterCount;
+
+        System.out.println("Total Weight:"+totalWeight + " scatterCount: " + scatterCount + " target:" + targetWeightPerScatter);
+
+        final List<IntervalList> scattered = new ArrayList<>();
+
+        float cumulativeWeight = 0;
+        IntervalList currentList = new IntervalList(sequenceDictionary);
+
+        PushbackIterator<Interval> iter = new PushbackIterator<>(intervalsWithWeights.iterator());
+        while (iter.hasNext()) {
+            WeightedInterval wi = (WeightedInterval) iter.next();
+
+            // if the interval fits completely, just add it
+            if (cumulativeWeight + wi.getWeight() <= targetWeightPerScatter ) {
+                cumulativeWeight += wi.getWeight();
+                currentList.add(wi);
+
+            // if it would push us over the edge
+            } else {
+                // add a piece of it
+                float remainingSpace = targetWeightPerScatter - cumulativeWeight;
+
+                // how many bases can we take?
+                int basesToTake = (int) Math.floor(remainingSpace /  wi.getWeightPerBase());
+
+                // split and add the first part into this list
+                WeightedInterval[] pair = wi.split(basesToTake);
+                currentList.add(pair[0]);
+
+                // push the remainder back onto the iterator
+                iter.pushback(pair[1]);
+
+                // add uniqued, sorted output list and reset
+                scattered.add(currentList.uniqued().sorted());
+                cumulativeWeight = 0;
+                currentList = new IntervalList(sequenceDictionary);
+            }
+        }
+        // add the final list
+        scattered.add(currentList.uniqued().sorted());
```

Comment:
This is making a copy of the header for every list in the scatter and saving it, which will add up really fast.